data/sea/54-zebra.h

```diff
@@ -0,0 +1,236 @@
+  , iint_t elem_start
+  , iint_t count
+  , void **dst
+
+  , const zebra_column_t *src );
+
+static ierror_msg_t zebra_read_entity (zebra_state_t *state, zebra_entity_t *entity);
+
+
+static void *zebra_copy_array (imempool_t *into, void *src, int64_t nbytes)
+{
+    void *alloc = imempool_alloc (into, nbytes);
+    memcpy (alloc, src, nbytes);
+    return alloc;
+}
+
+/* map a zebra entity to sea fleet inputs */
+static ierror_msg_t zebra_translate
+    ( imempool_t *mempool
+    , int         attribute_ix
+    , itime_t     chord_time
+    , iint_t     *new_count
+    , ierror_t   **new_tombstone
+    , itime_t    **new_fact_time
+    , void       **dst
+    , iint_t      columns_to_fill
+    , zebra_entity_t *src)
+{
+    zebra_attribute_t *attribute = &src->attributes[attribute_ix];
+    iint_t columns_count = attribute->table.column_count;
+
+    if (columns_count != columns_to_fill) {
+        return ierror_msg_format ("zebra_translate: attribute_ix=%d, column_count=%d, columns_to_fill=%d\n", attribute_ix, columns_count, columns_to_fill);
+     }
+
+    zebra_table_t *table = &attribute->table;
+    iint_t row_count = table->row_count;
+    iint_t fact_count = 0;
+
+    for (iint_t i = 0; i != row_count; ++i) {
+        itime_t time = attribute->times[i];
+        if (chord_time <= time) {
+            break;
+        }
+        fact_count++;
+    }
+
+    *new_count = fact_count;
+    *new_tombstone = (ierror_t*)attribute->tombstones;
+    *new_fact_time = attribute->times;
+
+    zebra_translate_table (mempool, 0, fact_count, dst, &attribute->table);
+
+    return 0;
+}
+
+static int64_t zebra_translate_table
+    ( imempool_t *mempool
+    , iint_t elem_start
+    , iint_t count
+    , void **dst
+    , const zebra_table_t *table )
+{
+    int64_t         offset = 0;
+    int64_t         cols_count  = table->column_count;
+    zebra_column_t *cols = table->columns;
+
+    for (int64_t i = 0; i != cols_count; ++i) {
+        offset += zebra_translate_column (mempool, elem_start, count, dst + offset, cols + i);
+    }
+
+    return offset;
+}
+
+static int64_t zebra_translate_column
+    ( imempool_t *mempool
+    , iint_t elem_start
+    , iint_t count
+    , void **dst
+    , const zebra_column_t *src )
+{
+    const zebra_type_t type = src->type;
+    const zebra_data_t data = src->data;
+
+    switch (type) {
+        case ZEBRA_BYTE:
+            {
+                uint8_t* bytes = imempool_alloc(mempool, count + 1);
+                memcpy(bytes, data.b + elem_start, count);
+                bytes[count] = '\0';
+                *dst = bytes;
+                return 1;
+            }
+
+        case ZEBRA_INT:
+            *dst = data.i + elem_start;
+            return 1;
+
+        case ZEBRA_DOUBLE:
+            *dst = data.d + elem_start;
+            return 1;
+
+        case ZEBRA_ARRAY:
+            {
+                istring_t *strings = imempool_alloc (mempool, count * sizeof (istring_t));
+                int offset = 1;
```

Comment:
It might be worth having a comment as to why this gets set to 1, and what the rough limitations are